app/[locale]/signin/components/signInForm.tsx

```diff
@@ -0,0 +1,117 @@
+'use client';
+
+import { cn } from '@/lib/utils';
+import { Button } from '@/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import { Input } from '@/components/ui/input';
+import { Label } from '@/components/ui/label';
+import { ComponentProps, useState } from 'react';
+import { useTranslations } from 'next-intl';
+
+import { getAuthSchema } from '@/lib/schemas/auth';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { useForm } from 'react-hook-form';
+import { signIn } from 'next-auth/react';
+import { useRouter } from 'next/navigation';
+
+export function SignInForm({ className, ...props }: ComponentProps<'div'>) {
+  const text = useTranslations();
+  const schema = getAuthSchema(text);
+  type FormData = z.infer<typeof schema>;
+  const [isLoading, setIsLoading] = useState(false);
+  const [serverError, setServerError] = useState<string | null>(null);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<{ email: string; password: string }>({
+    resolver: zodResolver(schema),

```

Comment:
Добавь еще mode, чтобы формы валидировались на лету, а не только после сабмита. Это необязательно по тз, но так проще проверять.
```suggestion
    resolver: zodResolver(schema),
    mode: 'onChange',
```

Reply:
исправлено